src/Features/Core/Portable/InlineMethod/AbstractInlineMethodRefactoringProvider.InlineContext.cs

```diff
@@ -0,0 +1,582 @@
+                    methodParametersInfo.OperationsNeedGenerateDeclarations.SelectAsArray(operation => operation.Parameter),
+                    parametersWithVariableDeclarationArgumentToName!,
+                    cancellationToken);
+
+                // Generate all the statements need to be put in the caller.
+                // Use the parameter's name to generate declarations in caller might cause conflict in the caller,
+                // so the rename table is needed to provide with a valid name.
+                // Example:
+                // Before:
+                // void Caller(int i, int j)
+                // {
+                //     Callee(Foo(), Bar());
+                // }
+                // void Callee(int i, int j)
+                // {
+                //     DoSomething(i, j);
+                // }
+                // After:
+                // void Caller(int i, int j)
+                // {
+                //     int i1 = Foo();
+                //     int j1 = Bar();
+                //     DoSomething(i1, j1)
+                // }
+                // void Callee(int i, int j)
+                // {
+                //     DoSomething(i, j);
+                // }
+                // Here two declaration is generated.
+                // Another case is
+                // void Caller()
+                // {
+                //     Callee(out int i)
+                // }
+                // void Callee(out int j)
+                // {
+                //     DoSomething(out j, out int i);
+                // }
+                // After:
+                // void Caller()
+                // {
+                //     int i = 10;
+                //     DoSomething(out i, out int i2);
+                // }
+                // void Callee(out int j)
+                // {
+                //     DoSomething(out j, out int i);
+                // }
+                var localDeclarationStatementsNeedInsert = GetLocalDeclarationStatementsNeedInsert(
+                    inlineMethodRefactoringProvider,
+                    syntaxGenerator,
+                    methodParametersInfo.OperationsNeedGenerateDeclarations,
+                    parametersWithVariableDeclarationArgumentToName!,
+                    renameTable);
+
+                // Get a table which the key is the symbol needs replacement work. Value is the replacement syntax node
+                var replacementTable = ComputeReplacementTable(
+                    inlineMethodRefactoringProvider,
+                    calleeMethodSymbol,
+                    methodParametersInfo.ParametersWithLiteralArgument,
+                    methodParametersInfo.ParametersWithDefaultValue,
+                    syntaxGenerator,
+                    renameTable);
+
+                // Do the replacement work within the callee's body so that it can be inserted to the caller later.
+                inlineNode = (TExpression)await ReplaceAllSyntaxNodesForSymbolAsync(
```

Comment:
why cast here?  why does ReplaceAllSyntaxNodesForSymbolAsync not already return a TExpression?

Reply:
Hmm, I am calling a syntax editor within it, so it is return syntax node. Does it has a generics version?

Reply:
SyntaxEditro may return a SyntaxNode, but ReplaceAllSyntaxNodesForSymbolAsync can then cast this itself.  in other words, push this cast in deeper inside your own feature.